lib/services/elasticsearch.js

```diff
@@ -473,6 +482,10 @@ class ElasticSearch extends Service {
     assertNoRouting(esRequest);
     assertWellFormedRefresh(this.config.apiVersion, esRequest);
 
+    if (!this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)) {
```

Comment:
Yes but I thought it was a good idea to notify the user the problem can comes from a typo or something in the index/collection and not from the rest of the request.
I prefer if the error is more precise about what the problem is. What do you think?

And the check is fast